app/views/api/v2/hosts/base.json.rabl

```diff
@@ -1,3 +1,11 @@
 object @host
 
-attributes :name, :id
+attributes :id
+
+node :name do |name|
```

Comment:
That's not quite good. This can potentially brake a lot of automation, since we change old API responses. That's what we should avoid.

Also, I don't think the setting should affect this. Since API response shows stored data and we are going to store the whole fqdn, this is not even needed.

Reply:
> Also, I don't think the setting should affect this. Since API response shows stored data and we are going to store the whole fqdn, this is not even needed.

It is needed because as you say we are going to store the whole FQDN. So without this change the host detail would always display the whole FQDN instead of a short name.

> That's not quite good. This can potentially brake a lot of automation, since we change old API responses. That's what we should avoid.

I need a way to cut the domain from FQDN stored in database before the data is sent to React. If you have suggestion how to do it differently, I will gladly chage it.

Reply:
API is not meant to be used by React only, users can directly call it through hammer or process response through other scripts.

You can simply cut the hostname the same way you did here, but within React component just using javascript.